stix_shifter/stix_transmission/src/modules/aws_cloud_watch_logs/aws_cloud_watch_logs_results_connector.py

```diff
@@ -0,0 +1,146 @@
+                                         "detail_service_eventLastSeen"]
+                self.format_results(guardduty_common_attr, result_list, results, return_obj)
+            else:
+                raise InvalidParameterException
+        except Exception as ex:
+            if isinstance(ex, InvalidParameterException):
+                return_obj = dict()
+                response_dict['__type'] = 'InvalidParameterException'
+                response_dict['message'] = 'Total number of records (offset+length) must be less than or equal to 10000'
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+            else:
+                response_dict['__type'] = ex.__class__.__name__
+                response_dict['message'] = ex
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'])
+        return return_obj
+
+    def format_results(self, guardduty_common_attr, result_list, results, return_obj):
+        """
+        Formatting the results
+        :param guardduty_common_attr: list, list of guardduty common attributes
+        :param result_list: list
+        :param results: list, results
+        :param return_obj: dict
+        :return: dict
+        """
+        for record in results:
+            record_dict = dict()
+            for data in record:
+                record_dict[data['field']] = data['value']
+            if 'source' in record_dict.keys() and record_dict['source'] == 'aws.guardduty':
+                json_message = record_dict['@message']
+                data = json.loads(json_message)
+                flatten_results = flatten(data)
+                flatten_results = {k: v for k, v in flatten_results.items() if v != "" and v != {}}
+                guard_dict = dict()
+                guard_dict['guardduty'] = dict()
+                if flatten_results.get('detail_service_action_actionType') is None:
+                    continue
+                if flatten_results.get('detail_service_action_networkConnectionAction_protocol') == 'Unknown':
+                    continue
+                self.process_flatten_results(flatten_results, guard_dict, guardduty_common_attr, record_dict)
+                result_list.append(guard_dict)
+            elif 'source' not in record_dict.keys():
+                vpc_dict = dict()
+                vpc_dict['vpcflow'] = copy.deepcopy(record_dict)
+                vpc_dict['vpcflow']['protocol'] = self.get_protocol(vpc_dict['vpcflow']['protocol'])
+                vpc_dict['vpcflow']['event_count'] = 1
+                result_list.append(vpc_dict)
+            else:
+                json_message = record_dict['@message']
+                data = json.loads(json_message)
+                flatten_results = flatten(data)
+                result_list.append(flatten_results)
+        return_obj['data'] = result_list
+
+    def process_flatten_results(self, flatten_results, guard_dict, guardduty_common_attr, record_dict):
```

Comment:
is the function `process_flatten_results()` only used for processing guard duty logs? If yes than you can add guardduty in the function name to make it more clear like `process_flatten_guardduty_results`. if we need similar function in future for a different aws logs, we can have separate function and also we can make a common utility function later.

Reply:
done